packages/astro/src/jsx/server.ts

```diff
@@ -48,8 +39,27 @@ export async function renderToStaticMarkup(
 	}
 
 	const { result } = this;
-	const html = await renderJSX(result, jsx(Component, { ...props, ...slots, children }));
-	return { html };
+	try {
+		const html = await renderJSX(result, jsx(Component, { ...props, ...slots, children }));
+		return { html };
+	} catch (e) {
+		throwEnhancedErrorIfMdxComponent(e as Error, Component);
+		throw e;
+	}
+}
+
+function throwEnhancedErrorIfMdxComponent(error: Error, Component: any) {
+	// if the exception is from an mdx component
+	// throw an error
+	if (Component[Symbol.for('mdx-component')]) {
```

Comment:
We started to move those symbols inside [`constants.ts`,](https://github.com/withastro/astro/blob/main/packages/astro/src/core/constants.ts) so we could document them. Does it make sense for this symbol to be placed together with the rest?